proposals/spline-animation/README.md

```diff
@@ -0,0 +1,1980 @@
+
+protected:
+    // Virtual dispatch for non-virtual interface.
+    // This is an implementation detail; these are possible examples.
+    virtual bool _SetPostInterp(TsInterpMethod method) = 0;
+    virtual bool _IsC0Continuous() const = 0;
+    virtual bool _IsG1Continuous() const = 0;
+    virtual bool _IsC1Continuous() const = 0;
+};
+
+class TsSplineKnot :
+    public TsKnotInterface
+{
+    // SetPostInterp: supports all methods.
+
+    void SetPreTanLen(double len);
+    double GetPreTanLen() const;
+
+    // The many tangent methods cover:
+    // - Setting and getting
+    // - Pre-tangents and post-tangents
+    // - Typed and VtValue
+    // - Presto and Maya forms
+    // - Auto-tangents
+    template <typename T> void SetPreTanSlope(const T &slope);
+    template <typename T> bool GetPreTanSlope(T *slopeOut) const;
+    void SetPreTanSlope(const VtValue &slope);
+    bool GetPreTanSlope(VtValue *slopeOut) const;
+    template <typename T> void SetPreTanHeight(const T &height);
+    template <typename T> bool GetPreTanHeight(T *heightOut) const;
+    void SetPreTanHeight(const VtValue &height);
+    bool GetPreTanHeight(VtValue *heightOut) const;
+    void SetPreTanAuto(bool auto);
+    bool IsPreTanAuto() const;
+    template <typename T> void SetPostTanSlope(const T &slope);
+    template <typename T> bool GetPostTanSlope(T *slopeOut) const;
+    void SetPostTanSlope(const VtValue &slope);
+    bool GetPostTanSlope(VtValue *slopeOut) const;
+    template <typename T> void SetPostTanHeight(const T &height);
+    template <typename T> bool GetPostTanHeight(T *heightOut) const;
+    void SetPostTanHeight(const VtValue &height);
+    bool GetPostTanHeight(VtValue *heightOut) const;
+    void SetPostTanAuto(bool auto);
+    bool IsPostTanAuto() const;
+};
+
+class TsLerpSeriesKnot :
+    public TsKnotInterface
+{
+    // SetPostInterp: supports Held and Linear.
+};
+
+class TsHeldSeriesKnot :
+    public TsKnotInterface
+{
+    // SetPostInterp: supports only Held.
+};
+
+class TsQuatSeriesKnot :
+    public TsKnotInterface
+{
+    // SetPostInterp: supports Held and Linear.
+};
+```
+
+All knots must have the same value type.  Attempting to mix knot types in the
+same series is an error.  Attempting to use an unsupported value type (e.g. for
+`TsSpline`, something other than `double`, `float`, or `half`) is also an error.
+
+There may only be one knot at any given time.  Passing multiple knots with the
+same time to `SwapKnots` is an error.  Calling `SetKnot` with an existing knot
+time will silently overwrite the existing knot.
+
+See [Question: time snapping](#time-snapping).
+
+## Automatic Tangent Computation
+
+When splines contain automatic tangents, these are recomputed each time nearby
+knots change.
+
+During bulk editing operations, automatic tangent computation can be deferred in
+order to avoid redundant processing.  An RAII helper class called
+`TsAutomaticTangentBlock` will be provided for this purpose.
```

Comment:
Realtime feedback in curve editing operations over many knots is a big concern of ours, we would make use of this.